pkg/apis/cluster/v1alpha1/cluster_consts.go

```diff
@@ -37,6 +37,8 @@ const (
 
 	// ReadyCondition defines the Ready condition type that summarizes the operational state of a Cluster.
 	ReadyCondition capiv1.ConditionType = "Ready"
+	// ProviderFailedReason (Severity=Error) documents that the cluster infrastructure provider experienced an error.
+	ProviderFailedReason = "ProviderFailed"
```

Comment:
It is not explicit what does it mean, keep in mind all these are to endusers, they should be able to understand